packages/runtime/container-runtime/src/test/blobManager.spec.ts

```diff
@@ -251,12 +257,44 @@ describe("BlobManager", () => {
 
         await createBlob(IsoBuffer.from("blob", "utf8"));
         await runtime.processAll();
-
         const summaryData = validateSummary(runtime);
         assert.strictEqual(summaryData.ids.length, 1);
         assert.strictEqual(summaryData.redirectTable, undefined);
     });
 
+    it("hasPendingBlobs()", async () => {
+        await runtime.attach();
```

Comment:
It would be good to add coverage for detached, offline, online->offline transition. I'm not sure if it will behave as you expect it to when detached.

Reply:
What's the expectation of how this signal will work in detached containers? I think it would be good to say explicitly in your design doc. It looks like `hasPendingBlobs` won't ever be true in detached since we don't add entries to the pending map. I think it would make sense to consider blobs added to detached containers as pending since they are stored locally and uploaded when the container is attached.